supporting_scripts/analysis-of-endpoint-connections/src/main/java/de/tum/cit/endpointanalysis/EndpointAnalyzer.java

```diff
@@ -60,10 +60,13 @@ private static void analyzeEndpoints() {
                 for (EndpointInformation endpoint : endpointClass.endpoints()) {
 
                     String endpointURI = endpoint.buildComparableEndpointUri();
-                    List<RestCallInformation> matchingRestCalls = restCallMap.getOrDefault(endpointURI, new ArrayList<>());
+                    List<RestCallInformation> restCallsWithMatchingURI = restCallMap.getOrDefault(endpointURI, new ArrayList<>());
 
                     // Check for wildcard endpoints if no exact match is found
-                    checkForWildcardEndpoints(endpoint, matchingRestCalls, endpointURI, restCallMap);
+                    checkForWildcardEndpoints(endpoint, restCallsWithMatchingURI, endpointURI, restCallMap);
+
+                    List<RestCallInformation> matchingRestCalls = restCallsWithMatchingURI.stream()
+                            .filter(restCall -> restCall.method().toLowerCase().equals(endpoint.getHttpMethod().toLowerCase())).toList();
```

Comment:
Are different capitalisations of the URIs actually the same?

Reply:
No, different capitalizations of URIs are different. What is compared here is the HTTP Method. On the client, it is specified with `httpClient.get(...)`, while on the Server it is specified with annotations like `@getMapping(...)`.